src/main/java/com/auctionapp/api/service/BidService.java

```diff
@@ -0,0 +1,99 @@
+package com.auctionapp.api.service;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import com.auctionapp.api.model.dto.BidDto;
+import com.auctionapp.api.model.entities.Bid;
+import com.auctionapp.api.repository.BidRepository;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class BidService {
+
+    private final BidRepository bidRepository;
+	private final AuctionService auctionService;
+
+    public BidService(final BidRepository bidRepository,
+					  final AuctionService auctionService) {
+        this.bidRepository = bidRepository;
+		this.auctionService = auctionService;
+    }
+
+    public List<BidDto> getAuctionBids(final UUID auctionId) {
+        final List<Bid> bids = bidRepository.findAllByAuctionId(auctionId);
+		return bids.stream().map(t -> toPayload(t)).collect(Collectors.toList());
+    }
+
+	public BidDto save(final BidDto payload) {
+		if (existsAmongBidders(payload.getBuyer().getId(), payload.getAuction().getId())) {
+			return toPayload(update(payload));
+		}
+
+		payload.setBidDate(Timestamp.from(Instant.now()));
+        Bid bid = fromPayload(payload);
+        bid = bidRepository.save(bid);
+		bid.setAuction(auctionService.update(payload.getAuction(), payload.getBidAmount()));
+        return toPayload(bid);
+	}
+
+	public Bid update(final BidDto payload) {
+        Optional<Bid> bidEntity = bidRepository.findByBuyerIdAndAuctionId(payload.getBuyer().getId(), payload.getAuction().getId());
+
+        if (bidEntity.isPresent()) {
+			if (payload.getBidAmount() > bidEntity.get().getBidAmount()) {
+				bidEntity.get().setBidAmount(payload.getBidAmount());
+				bidEntity.get().setBidDate(Timestamp.from(Instant.now()));
+				Bid  bid = bidRepository.save(bidEntity.get());
+				bid.setAuction(auctionService.update(payload.getAuction(), payload.getBidAmount()));
+				return bid;
+			}
+			throw new RuntimeException("New bid amount has to be greater than the previous ones!");
+        }
+        throw new RuntimeException("Bid with id " + payload.getId() + " does not exist!");
+    }
```

Comment:
A bidder can place multiple bids on an auction. If you and I were outbidding each other on some item, we want the history of who bid when preserved. All that to say - you should store each bid as a separate entry and keep a list of them in the auction, like I mentioned in the previous PR.

Reply:
Now that Nejra requested only seller to be able to see the list of bidders, should I do it like this, for the same bidder to make a new record each time they make a bid on a certain item, rather than update the previous bid value for that person with their new bid value?

Reply:
Each bid should still be a separate entry. You generally do not want to overwrite/lose the history of client interactions with your system, especially when it comes to something important (if these were to be actually used by clients, we'd be talking about money and potentially highly sought-after items). If bugs were to occur, and they always do, you want to have as much detail as possible about the interactions that led to them.

Reply:
Aha, makes sense. It is implemented now like that